### DISCUSSION TITLE
Add "Singular" Name Fields to Collection Settings

### DISCUSSION THREAD

**Discussion by benhaynes** [directus/directus#3069] [Feature Requests]
Add singular “Item Unit Name” and “Item Unit Icon” fields to Collection Settings... we can more accurately refer to individual (vs plural) items in human terms.

**Comment by abdonrd** [comment]
The current Directus recommendation is to use the plural for the collection name?

So `countries` instead of `country`?

| `countries` | `country` |
|-------------|-----------|
| <img width="856" alt="Screenshot 2022-01-30 at 12 55 23" src="https://user-images.githubusercontent.com/1007051/151698588-833be7d4-428f-46a2-bd8a-dd4bda5fc87d.png"> | <img width="857" alt="Screenshot 2022-01-30 at 12 55 34" src="https://user-images.githubusercontent.com/1007051/151698594-8464f15a-33f5-4c47-a94b-74c4ca46984f.png"> |

**Reply by rijkvanzanten** [reply]
It's totally up to you what you want to use for the collection key (== table name). You can use translations to show it differently in the app 👍🏻

**Comment by KrisKnez** [comment]
This is the table naming dilemma. Read more here: https://stackoverflow.com/questions/338156/table-naming-dilemma-singular-vs-plural-names

I think this is a good feature request, some other CMS systems do require you or generate themselves the plural form of your table name.

**Reply by rijkvanzanten** [reply]
Yup! Directus already supports this currently 👍🏻 You can make the collection name whatever you want, and then use translations to show it differently in the app

**Reply by wilkomanger** [reply]
It does seem that Directus sometimes still assumes plural names, for example when using Translations, it assumes by default the language table is called `languages`, and even if the table is called `language`, it will still name the foreign key `languages_id`

**Reply by rijkvanzanten** [reply]
@wilkomanger That's just because the translations interface defaults to those names for ease of configuration, you're not required to use those 👍🏻

**Reply by wilkomanger** [reply]
Is that also true for the last example? It seems that if I add a Translations field, and I don't have a language table yet but fill in the `language` name, it will create the `language` table correctly. However the foreign key on e.g. `test_translations` is still called `languages_id`, which cannot be changed in Directus:

![image](https://user-images.githubusercontent.com/51335189/158193865-929f2dfa-328d-4eee-b253-074069b6d279.png)

**Reply by wilkomanger** [reply]
Or with an even more different language table name:
![image](https://user-images.githubusercontent.com/51335189/158194362-baa0a159-b7bd-4a8b-82e1-4362a0d9cc71.png)

I would expect `i18n_id` here :thinking:

**Reply by rijkvanzanten** [reply]
Yeah, good one 👍🏻 Seems like updating the languages_collection for translations doesn't correctly auto-update the auto-fill value for the m2o field to languages. Mind opening a new issue?

**Reply by wilkomanger** [reply]
[Done!](https://github.com/directus/directus/issues/12255)
